node/src/protocol/mod.rs

```diff
@@ -255,6 +256,9 @@ impl MpcSignProtocol {
                 let guard = self.state.read().await;
                 guard.clone()
             };
+
+            let before_step = SystemTime::now();
```

Comment:
we should use `Instant::now()` and then call into `elapsed() < Duration::from_millis(10)` instead since that's more readable. 10ms is also for good measure since the amount of time it takes can vary based on different machines. Maybe even 50ms is better option.

Reply:
Good point, changed